types/options.go

```diff
@@ -273,10 +275,16 @@ func isRootlessDriver(driver string) bool {
 func getRootlessStorageOpts(rootlessUID int, systemOpts StoreOptions) (StoreOptions, error) {
 	var opts StoreOptions
 
-	dataDir, rootlessRuntime, err := getRootlessDirInfo(rootlessUID)
+	dataDir, err := homedir.GetDataHome()
```

Comment:
This makes the `rootlessUID` value ignored, and that propagates up the call stack… `DefaultStoreOptions` is widely called, and it has at least one caller who calls it with `(false, 0)`.

I would love for the configuration loading code of c/storage to be simplified, but I quickly gave up every time I looked at that spaghetti; doing that in this PR seems way too big a scope expansion.

Reply:
`getRootlessDirInfo(rootlessUID int) (string, string, error)` also ignored it:

```
      rootlessRuntime, err := GetRootlessRuntimeDir(rootlessUID)
      if err != nil {
               return "", "", err
      }

      home := homedir.Get()
       if home == "" {
               return "", "", fmt.Errorf("neither XDG_DATA_HOME nor HOME was set non-empty: %w", err)
       }
```

only `GetRootlessRuntimeDir()` used it to hardcode in the directory path.

The caller with `(false, 0)` uses only the UIDMap and GIDMap settings so it should not be affected by this change.  I'll vendor it in Podman to see if anything breaks.

Reply:
> `getRootlessDirInfo(rootlessUID int) (string, string, error)` also ignored it:

That’s not what I see at all?! `rootlessRuntime` is returned on every non-error exit from `getRootlessDirInfo`.

Reply:
> The caller with `(false, 0)` uses only the UIDMap and GIDMap settings so it should not be affected by this change.

Good point.

Also, looking at `DefaultStoreOptions`, it makes no sense at all to have the two parameters, but then process them only under a `sync.Once`. So I think that should be deprecated in favor of some variant of `DefaultStoreOptionsAutoDetectUID`.

But that is really looking like 2 or 3 separate PRs…